vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/TextField.java

```diff
@@ -337,4 +361,167 @@ protected void onAttach(AttachEvent attachEvent) {
         super.onAttach(attachEvent);
         ClientValidationUtil.preventWebComponentFromModifyingInvalidState(this);
     }
+
+    /**
+     * Gets the internationalization object previously set for this component.
+     * <p>
+     * NOTE: Updating the instance that is returned from this method will not
+     * update the component if not set again using
+     * {@link #setI18n(TextFieldI18n)}
+     *
+     * @return the i18n object or {@code null} if no i18n object has been set
+     */
+    public TextFieldI18n getI18n() {
+        return i18n;
+    }
+
+    /**
+     * Sets the internationalization object for this component.
+     *
+     * @param i18n
+     *            the i18n object, not {@code null}
+     */
+    public void setI18n(TextFieldI18n i18n) {
+        this.i18n = Objects.requireNonNull(i18n,
+                "The i18n properties object should not be null");
+    }
+
+    private String getRequiredErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(TextFieldI18n::getRequiredErrorMessage).orElse("");
+    }
+
+    private String getMinLengthErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(TextFieldI18n::getMinLengthErrorMessage).orElse("");
+    }
+
+    private String getMaxLengthErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(TextFieldI18n::getMaxLengthErrorMessage).orElse("");
+    }
+
+    private String getPatternErrorMessage() {
+        return Optional.ofNullable(i18n)
+                .map(TextFieldI18n::getPatternErrorMessage).orElse("");
+    }
```

Comment:
Could use the same approach as suggested in https://github.com/vaadin/flow-components/pull/6365#discussion_r1675358980

Reply:
Updated.